gce-containers-startup/utils/volumes.go

```diff
@@ -174,20 +184,30 @@ func processGcePersistentDiskVolume(volume *api.GcePersistentDiskVolume) (Volume
 	if err := checkDeviceNotMounted(devicePath); err != nil {
 		return VolumeHostPathAndMode{}, err
 	}
-	if err := checkFilesystemAndFormatIfNeeded(devicePath, volume.FsType); err != nil {
-		return VolumeHostPathAndMode{}, err
+	if !readOnly {
+		if err := checkFilesystemAndFormatIfNeeded(devicePath, volume.FsType); err != nil {
+			return VolumeHostPathAndMode{}, err
+		}
 	}
 
-	deviceMountPoint, err := createNewMountPath("gce_persistent_disk", volume.Name)
+	deviceMountPoint, err := createNewMountPath("gce_persistent_disk", volume.PdName)
 	if err != nil {
 		return VolumeHostPathAndMode{}, err
 	}
-	if err := mountDevice(devicePath, volume.ReadOnly); err != nil {
+	if err := mountDevice(devicePath, deviceMountPoint, volume.FsType, readOnly); err != nil {
 		return VolumeHostPathAndMode{}, err
 	}
 
 	// Success!
-	return VolumeHostPathAndMode{deviceMountPoint, volume.ReadOnly}, nil
+	return VolumeHostPathAndMode{deviceMountPoint, readOnly}, nil
+}
+
+// Determine whether the GCE Persistent Disk is attached RO or RW.
+func checkIfGcePersistentDiskIsReadOnly(pdName string) (readOnly bool, err error) {
+	// TODO: Implement reading the metadata.
+	err = nil
+	readOnly = false
+	return
```

Comment:
Why not return (err, readOnly) or return (nil, false) to be consistent across the file?

Reply:
I have removed this function altogether in a later commit.